### PR TITLE
[mypy] [9889] Reduce mypy errors

### PR DESCRIPTION
**Pull request by rodrigc**
https://twistedmatrix.com/trac/ticket/9889

### CODE REVIEW

File: src/twisted/internet/test/test_epollreactor.py [hunk 1]
```diff
@@ -8,9 +8,9 @@
 
 from twisted.trial.unittest import TestCase
 try:
-    from twisted.internet.epollreactor import _ContinuousPolling
+    from twisted.internet.posixbase import _ContinuousPolling
```
Comment (adiroiban) [inline src/twisted/internet/test/test_epollreactor.py]: I think that we should have epollreactor ... this is a test for epollreactor so I think that we should have an explicit import here.

File: src/twisted/_threads/test/test_team.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 from twisted.trial.unittest import SynchronousTestCase
 
-from twisted.python.context import call, get
+from twisted.python.context import call, get  # type: ignore[attr-defined]
```
Comment (adiroiban) [inline src/twisted/_threads/test/test_team.py]: I don't know to say about t.p.context.py

I have no idea why installContextTracker is a public API.

@glyph do you see any issue with the following diff  which removes installContextTracker and the use of global

```diff
diff --git a/src/twisted/python/context.py b/src/twisted/python/context.py
index 1348822e1..e208e3dc2 100644
--- a/src/twisted/python/context.py
+++ b/src/twisted/python/context.py
@@ -22,7 +22,6 @@ defaultContextDict = {}  # type: Dict[Type[object], Dict[str, str]]
 setDefault = defaultContextDict.__setitem__
 
 
-
 class ContextTracker:
     """
     A L{ContextTracker} provides a way to pass arbitrary key/value data up and
@@ -124,13 +123,7 @@ class ThreadedContextTracker(object):
         return self.currentContext().getContext(key, default)
 
 
-def installContextTracker(ctr):
-    global theContextTracker
-    global call
-    global get
-
-    theContextTracker = ctr
-    call = theContextTracker.callWithContext
-    get = theContextTracker.getContext
 
-installContextTracker(ThreadedContextTracker())
+theContextTracker = ThreadedContextTracker()
+call = theContextTracker.callWithContext
+get = theContextTracker.getContext
```

File: src/twisted/words/test/test_jabberjstrports.py [hunk 3]
```diff
@@ -9,7 +9,8 @@
 from twisted.trial import unittest
 
 from twisted.words.protocols.jabber import jstrports
-from twisted.application.internet import TCPClient
+from twisted.application.internet import TCPClient  # type: ignore[attr-defined]  # noqa
```
Comment (adiroiban) [inline src/twisted/words/test/test_jabberjstrports.py]: this is ok... the code from t.application.internet is ingenious  and we can keep it... I would vote for refactoring it to make it easier to read... for multicast and UDP we only have server-side applicitons.

below is the code from application/internet.py
```
_doc={
'Client':
"""Connect to %(tran)s

Call reactor.connect%(tran)s when the service starts, with the
arguments given to the constructor.
""",
'Server':
"""Serve %(tran)s clients

Call reactor.listen%(tran)s when the service starts, with the
arguments given to the constructor. When the service stops,
stop listening. See twisted.internet.interfaces for documentation
on arguments to the reactor method.
""",
}

for tran in 'TCP UNIX SSL UDP UNIXDatagram Multicast'.split():
    for side in 'Server Client'.split():
        if tran == "Multicast" and side == "Client":
            continue
        if tran == "UDP" and side == "Client":
            continue
        base = globals()['_Abstract'+side]
        doc = _doc[side] % vars()

        klass = type(tran+side, (base,), {'method': tran, '__doc__': doc})
        globals()[tran+side] = klass
```

**Review by adiroiban — COMMENTED** [review]
just a comment for now... but I would say that this require a change and have an explicit import for epollreactor

thanks

File: src/twisted/internet/test/test_epollreactor.py (hunk 1 above)
Comment (rodrigc) [reply src/twisted/internet/test/test_epollreactor.py]: OK, I added an explicit import of epollreactor to test_epollreactor.py

File: src/twisted/words/test/test_jabberjstrports.py (hunk 3 above)
Comment (rodrigc) [reply src/twisted/words/test/test_jabberjstrports.py]: OK, I removed the change for this PR, and refactored the code here: https://github.com/twisted/twisted/pull/1339

File: src/twisted/_threads/test/test_team.py (hunk 2 above)
Comment (rodrigc) [reply src/twisted/_threads/test/test_team.py]: OK, I removed changes from this PR, and created a new PR to address this: https://github.com/twisted/twisted/pull/1340

**Comment by rodrigc** [comment]
@adiroiban Can I merge this?  I incorporated your feedback.

**Review by adiroiban — APPROVED** [review]
All good. Thanks for the update!

**Pull request merged**